mypy/types.py

```diff
@@ -357,6 +357,7 @@ class UninhabitedType(Type):
     can_be_true = False
     can_be_false = False
     is_noreturn = False  # Does this come from a NoReturn?  Purely for error messages.
+    ambiguous = False  # Is this a result of inference for a variable without constraints?
```

Comment:
Also mention why we care about this and where this makes a difference.

Reply:
> Also mention why we care about this and where this makes a difference.

I added a short discussion.